### PR TITLE
add ep40 qmk support

### PR DESCRIPTION
**Pull request by e11i0t23**
Adds support for the ep40 keyboard along with a default keymap to allow for use. The firmware builds this correctly and has been tested to be working on the keyboard.

### CODE REVIEW

File: keyboards/handwired/ep40/info.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+    "keyboard_name":"ep40",
+    "url":null,
+    "maintainer":"e11i0t23",
+    "width":12,
+    "hight":4,
+    "layouts": {
+        "LAYOUT":{
+            "key_count":43,
+            "layout":[]
```
Comment (mechmerlin) [inline keyboards/handwired/ep40/info.json]: This is an incomplete info.json. You can make this by making a KLE of what your layout looks like, and then copy pasting the raw into https://qmk.fm/converter/

File: keyboards/handwired/ep40/readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# ep40
+
+![ep40](https://i.imgur.com/Jl7w6I9.jpg)
+
+A simple 40% keyboard
+
+Keyboard Maintainer: [Elliot Powell](https://reddit.com/u/e11i0t23)
```
Comment (mechmerlin) [inline keyboards/handwired/ep40/readme.md]: Could you also put your github account here too, in addition to your reddit one.

File: keyboards/handwired/ep40/rules.mk [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# MCU name
+#MCU = at90usb1286
```
Comment (mechmerlin) [inline keyboards/handwired/ep40/rules.mk]: You can remove this line 
```suggestion

```

File: keyboards/atreus62/info.json [hunk 4]
```diff
@@ -5,8 +5,9 @@
   "width": 15,
```
Comment (noroadsleft) [inline keyboards/atreus62/info.json]: This file is configured for Atreus62. If this is meant for the EP40, this file needs to be at:

`keyboards/handwired/ep40/info.json`

File: keyboards/atreus62/info.json [hunk 5]
```diff
@@ -5,8 +5,9 @@
...
   "height": 5.7,
   "layouts": {
+    "layouts": {
```
Comment (noroadsleft) [inline keyboards/atreus62/info.json]: This file's JSON is invalid. This line needs to be deleted.

```suggestion
```

File: keyboards/handwired/ep40/config.h [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+ *
+ * The options below allow the magic key functionality to be changed. This is
+ * useful if your keyboard/keypad is missing keys and you want magic key support.
+ *
+ */
+
+/* key combination for magic key command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
```
Comment (vomindoraan) [inline keyboards/handwired/ep40/config.h]: This is the default, so there's no need to define it.

Comment (e11i0t23) [reply keyboards/handwired/ep40/config.h]: Running the make command after removing this leads to an error however when I leave it in no errors occur

Comment (vomindoraan) [reply keyboards/handwired/ep40/config.h]: Ah okay, I figured there's a default define for this in core somewhere, but there might not be (we should probably add it at some point). All good then.

Comment (drashna) [reply keyboards/handwired/ep40/config.h]: @vomindoraan we should probably add a "ifndef" in the core code, for this, yeah.

**Comment by drashna** [comment]
Thanks!

File: keyboards/handwired/ep40/config.h (hunk 6 above)
Comment (vomindoraan) [reply keyboards/handwired/ep40/config.h]: @drashna Yep, then you wouldn't have to `#undef` either. The way it is atm isn't very clean. I'll open an issue for this.

**Pull request merged**